### PR TITLE
Code formatting changes following internal code review

### PR DESCRIPTION
**Pull request by hardbyte**
Changes from Brendon after running an internal code review.

### CODE REVIEW

File: clkhash/field_formats.py [hunk 1]
```diff
@@ -325,23 +325,25 @@ def __init__(self,
...
         if regex_based:
             regex_str = cast(str, regex)
             try:
                 compiled_regex = re_compile_full(regex_str)
+                self.regex = compiled_regex
             except (SyntaxError, re.error) as e:
                 msg = "invalid regular expression '{}.'".format(regex_str)
                 e_new = InvalidEntryError(msg)
                 e_new.field_spec = self
                 raise_from(e_new, e)
```
Comment (nbgl) [inline clkhash/field_formats.py]: Why this change? Does PyCharm not understand `raise_from`?

**Comment by wilko77** [comment]
I pushed some additional changes. 

Some of the proposed `noinspection` annotations do not seem to be needed any more. However, I found a few more issues where those are needed to pass the compatibility check.

So in total it is a net loss in terms of `# noinspection` pollution.

To make up for it I fixed some PIP violations and out-of-date docstrings.
